srml/sudo/src/lib.rs

```diff
@@ -14,8 +14,106 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! The Example: A simple example of a runtime module demonstrating
-//! concepts, APIs and structures common to most runtime modules.
+//! # Sudo Module
+//!
+//! ## Overview
+//! 
+//! The sudo module allows for a single account to be set as the sudo key of your blockchain. The sudo key can execute dispatchable functions which require a `Root` call or assign a new account to replace them as the sudo key.
+//! 
+//! You can start using the Sudo module by implementing the sudo [`Trait`].
+//! 
+//! Supported dispatchables functions are documented as part of the [`Call`] enum.
+//! 
+//! ## Interface
+//! 
+//! ### Dispatchable
+//! 
+//! Only the sudo key can call the dispatchable functions from the Sudo module.
+//! 
+//! * `sudo` - Make a `Root` call to a dispatchable function.
+//! * `set_key` - Assign a new account to be the key of the superuser
+//! 
+//! Please refer to the [`Call`] enum and its associated variants for a detailed list of dispatchable functions.
+//!
+//! ## Usage
+//! 
+//! ### Prerequisites
+//! 
+//! To use the sudo module in your runtime, you must implement the following trait in your runtime:
+//! 
+//! ```ignore
+//! impl sudo::Trait for Runtime {
+//! 	/// The uniquitous event type.
+//! 	type Event = Event;
+//! 	type Proposal = Call;
+//! }
+//! ```
+//! 
+//! You can then import the sudo module in your `construct_runtime!` macro with:
```

Comment:
Are we giving these types of instructions in the SRML level docs? If the runtime construction changes in the future, that will create a lot of rework. Could we have a single doc to explain how to build your runtime lib.rs?

Reply:
Agreed. I think though we should have some clarification with how you include the module. 
i.e.:

```
		Timestamp: timestamp::{Module, Call, Storage, Config<T>, Inherent},
		Consensus: consensus::{Module, Call, Storage, Config<T>, Log(AuthoritiesChange), Inherent},
		Aura: aura::{Module},
		Indices: indices,
		Balances: balances,
		Sudo: sudo,
Fees: fees::{Module, Storage, Config<T>, Event<T>},
```

It would be good to clarify which modules need which types defined in the construct_runtime macro, you know?

Reply:
Yeah, that's true. I'd say it's appropriate to document things that make this module different from another _SRML_ module.